docs/en/building/supplemental.md

```diff
@@ -0,0 +1,541 @@
+
+## Manifest
+
+The [manifest](https://developer.mozilla.org/en-US/docs/Web/Manifest) describes an application in a JSON file and provides details so it may be installed on a device's homescreen directly from the web.
+
+> .dojorc
+
+```json
+{
+	"build-app": {
+		"pwa": {
+			"manifest": {
+				"name": "Todo MVC",
+				"description": "A simple to-do application created with Dojo",
+				"icons": [
+					{ "src": "./favicon-16x16.png", "sizes": "16x16", "type": "image/png" },
+					{ "src": "./favicon-32x32.png", "sizes": "32x32", "type": "image/png" },
+					{ "src": "./favicon-48x48.png", "sizes": "48x48", "type": "image/png" },
+					{ "src": "./favicon-256x256.png", "sizes": "256x256", "type": "image/png" }
+				]
+			}
+		}
+	}
+}
+```
+
+When a manifest is provided `dojo build` will inject the necessary `<meta>` tags in the applications `index.html`.
+
+-   `mobile-web-app-capable="yes"`: indicates to Chrome on Android that the application can be added to the user's homescreen.
+-   `apple-mobile-web-app-capable="yes"`: indicates to iOS devices that the application can be added to the user's homescreen.
+-   `apple-mobile-web-app-status-bar-style="default"`: indicates to iOS devices that the status bar should use the default appearance.
+-   `apple-touch-icon="{{icon}}"`: the equivalent of the manifests' icons since iOS does not currently read icons from the manifest. A separate meta tag is injected for each entry in the icons array.
+
+## Service worker
+
+A service worker is a type of web worker that is able to intercept network requests, cache, and provide resources. Dojo's build command can automatically build fully-functional service worker that is activated on startup and complete with precaching and custom route handling from a configuration file.
+
+For instance, we could write a configuration to create a simple service worker that cached all of the application bundles except the admin bundle and cached recent application images and articles.
+
+> .dojorc
+
+```json
+{
+	"build-app": {
+		"pwa": {
+			"serviceWorker": {
+				"cachePrefix": "my-app",
+				"excludeBundles": ["admin"],
+				"routes": [
+					{
+						"urlPattern": ".*\\.(png|jpg|gif|svg)",
+						"strategy": "cacheFirst",
+						"cacheName": "my-app-images",
+						"expiration": { "maxEntries": 10, "maxAgeSeconds": 604800 }
+					},
+					{
+						"urlPattern": "http://my-app-url.com/api/articles",
+						"strategy": "cacheFirst",
+						"expiration": { "maxEntries": 25, "maxAgeSeconds": 86400 }
+					}
+				]
+			}
+		}
+	}
+}
+```
+
+### ServiceWorker Configuration
+
+Under the hood, the `ServicerWorkerPlugin` from `@dojo/webpack-contrib` is used to generate the service worker, and all of its options are valid `pwa.serviceWorker` properties.
+
+| Property       | Type       | Optional | Description                                                                                     |
```

Comment:
We'd previously been using unordered lists with abbreviated-TypeScript syntax to describe API parameter/config option lists ([for example](https://github.com/dojo/framework/blob/master/docs/en/i18n/supplemental.md#i18nmixin-widget-properties)), but I actually prefer this tabular layout; it's cleaner and more structured. We should probably revisit the committed ref guides and update them with this format for consistency (separate task to this ref guide of course!).